Content/Items/Misc/Weapons.Earthduster.cs

```diff
@@ -0,0 +1,560 @@
+                AddIngredient(ItemID.Bone, 50).
+                AddTile(TileID.Anvils).
+                Register();
+
+            CreateRecipe().
+                AddIngredient(ModContent.ItemType<Earthduster>()).
+                AddIngredient(ItemID.CrimtaneBar, 12).
+                AddIngredient(ItemID.Bone, 50).
+                AddTile(TileID.Anvils).
+                Register();
+        }
+    }
+
+    class EarthdusterHoldout : ModProjectile
+    {
+        public const int MAXSHOTS = 100;
+
+        public const int MAXCHARGEDELAY = 30;
+
+        public int SoilType;
+
+        public int shots;
+
+        public int initialShots;
+
+        public float rotTimer;
+
+        public bool charged;
+
+        public bool draw; //only draw two ticks after spawning
+
+        public bool reloading;
+
+        public bool forceReload;
+
+        public Vector2 mouse;
+
+        public ref float ShootDelay => ref Projectile.ai[0];
+
+        public ref float MaxShootDelay => ref Projectile.ai[1];
+
+        public Player owner => Main.player[Projectile.owner];
+
+        public bool CanHold => owner.channel && !owner.CCed && !owner.noItems;
+
+        public override string Texture => AssetDirectory.MiscItem + Name;
+
+        public override bool? CanDamage() => false;
+
+        public override void Load()
+        {
+            GoreLoader.AddGoreFromTexture<SimpleModGore>(Mod, Texture + "_Gore1");
+            GoreLoader.AddGoreFromTexture<SimpleModGore>(Mod, Texture + "_Gore2");
+        }
+
+        public override void SetStaticDefaults()
+        {
+            DisplayName.SetDefault("Earthduster");
+        }
+
+        public override void SetDefaults()
+        {
+            Projectile.DamageType = DamageClass.Ranged;
+
+            Projectile.width = 72;
+            Projectile.height = 62;
+            Projectile.friendly = true;
+            Projectile.tileCollide = false;
+            Projectile.ignoreWater = true;
+        }
+
+        public override void AI()
+        {
+            Vector2 armPos = owner.RotatedRelativePoint(owner.MountedCenter, true);
+            armPos += Utils.SafeNormalize(Projectile.velocity, Vector2.UnitX) * 20f;
+            armPos += Vector2.UnitY.RotatedBy(Projectile.velocity.ToRotation()) * -10f * owner.direction;
+            Vector2 barrelPos = armPos + (Projectile.velocity * 25f);
+
+            Vector2 dirtPos = armPos + new Vector2(-30, MathHelper.Lerp(-20f, -10f, shots / (float)MAXSHOTS) * owner.direction).RotatedBy(Projectile.rotation);
+
+            if (MaxShootDelay == 0f)
+                MaxShootDelay = CombinedHooks.TotalUseTime(owner.HeldItem.useTime, owner, owner.HeldItem);
+
+            if (rotTimer > 0)
+                rotTimer--;
+
+            if (!CanHold && !reloading && !forceReload)
+                forceReload = true;
+
+            if ((owner.HeldItem.ModItem as Earthduster).currentAmmoStruct.projectileID != SoilType)
```

Comment:
It would probably be a good move to introduce a nullable property for ``owner.HeldItem.ModItem as Earthduster`` and use null checking/operators for conciseness here